### PR TITLE
Remove ListItem base from Configuration

### PR DESCRIPTION
**Pull request by rprospero**
The main `List<Configuration>` was replaced with a `std::vector<std::unique_ptr<Configuration>>` while all of the `RefList<Configuration>` could then be converted to `std:vector<Configuration*>`.  This is another part of #257.

### CODE REVIEW

File: src/modules/analyse/process.cpp [hunk 1]
```diff
@@ -9,15 +9,12 @@
 bool AnalyseModule::process(Dissolve &dissolve, ProcessPool &procPool)
 {
     // Check for zero Configuration targets
-    if (targetConfigurations_.nItems() == 0)
+    if (targetConfigurations_.size() == 0)
```
Comment (trisyoungs) [inline src/modules/analyse/process.cpp:12]:
```suggestion
    if (targetConfigurations_.empty())
```

File: src/modules/calculate_axisangle/process.cpp [hunk 2]
```diff
@@ -15,7 +15,7 @@ bool CalculateAxisAngleModule::setUp(Dissolve &dissolve, ProcessPool &procPool)
 bool CalculateAxisAngleModule::process(Dissolve &dissolve, ProcessPool &procPool)
 {
     // Check for zero Configuration targets
-    if (targetConfigurations_.nItems() == 0)
+    if (targetConfigurations_.size() == 0)
```
Comment (trisyoungs) [inline src/modules/calculate_axisangle/process.cpp:18]:
```suggestion
    if (targetConfigurations_.empty())
```

File: src/modules/calculate_dangle/process.cpp [hunk 3]
```diff
@@ -15,7 +15,7 @@ bool CalculateDAngleModule::setUp(Dissolve &dissolve, ProcessPool &procPool) { r
 bool CalculateDAngleModule::process(Dissolve &dissolve, ProcessPool &procPool)
 {
     // Check for zero Configuration targets
-    if (targetConfigurations_.nItems() == 0)
+    if (targetConfigurations_.size() == 0)
```
Comment (trisyoungs) [inline src/modules/calculate_dangle/process.cpp:18]:
```suggestion
    if (targetConfigurations_.empty())
```

**Review by trisyoungs — APPROVED** [review]
A few missed opportunities to use `empty()` over `size() == 0`, but otherwise all good.

File: src/classes/coredata.cpp [hunk 4]
```diff
@@ -388,27 +388,32 @@ Species *CoreData::findSpecies(std::string_view name) const
...
 }
 
 // Remove specified Configuration
-void CoreData::removeConfiguration(Configuration *cfg) { configurations_.remove(cfg); }
+void CoreData::removeConfiguration(Configuration *cfg)
+{
+    auto it = std::find_if(configurations_.begin(), configurations_.end(), [cfg](const auto &c) { return cfg == c.get(); });
+    if (it != configurations_.end())
+        configurations_.erase(it);
+}
```
[the remark this answers is not in the document]
Comment (rprospero) [reply src/classes/coredata.cpp]: I had a reason, but, looking back on it, it wasn't a good one.  I've changed back to remove_if.

**Pull request merged**